### ISSUE TITLE
Buffer attach delta is mishandled

### ISSUE THREAD

**Issue by kennylevinsen** [swaywm/sway#6020]
Sway largely ignores the buffer attach point, and uses it to offset a surface away from its view origin, which is not correct behavior. This behavior can be observed by running the [wleird](https://github.com/emersion/wleird) `resizor` client.

Sway should instead use the buffer attach point delta to move floating windows, and ignore the buffer attach point otherwise.

It should be noted that the wlroots API here seems a bit awkward, as wlroots adds the deltas to x/y coordinates on the surface. To get the delta, which is the only interesting value, we must reset these coordinates to 0 on commit.

**Comment by emersion** [comment]
Yeah, the wlroots API is weird. It was supposed to simplify compositors, allowing them to just add the accumulated x/y attach coords to their final coordinates. This doesn't work well in practice.

I'd really want to avoid having the last attach x/y delta stored in `wlr_surface` though, since it's really a per-commit thing. We should probably expose it in the surface commit event struct.

**Comment by kennylevinsen** [comment]
> We should probably expose it in the surface commit event struct.

Agreed. I was thinking of maybe a dedicated surface move event, but adding to commit events works too.

I did get things working with the current API, but it's not very nice at all.
